src/plugins/credential_management/public/service/credential_management_service.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { HttpSetup } from '../../../../core/public';
+import { CredentialCreationConfig, CredentialCreationManager } from './creation';
+
+interface SetupDependencies {
+  httpClient: HttpSetup;
+}
+
+/**
+ * credential management service
+ *
+ * @internal
+ */
+export class CredentialManagementService {
+  credentialCreationManager: CredentialCreationManager;
```

Comment:
You could type this class as a Plugin, also, did you want to add the logger?

Reply:
> You could type this class as a Plugin, also, did you want to add the logger?

Removed this service for the initial PR. Also curious if we can add logger for brower side plugins.